Update Docker Documentation (#808)
wso2/docs-open-banking: en/docs/install-and-setup/deploy-with-docker.md

    2. Clone the [wso2/docker-open-banking](https://github.com/wso2/docker-open-banking.git) repository. This document 
       refers to the file path of the cloned directory as `<OB_DOCKER_HOME>`.

>     3. Download and extract the WSO2 IS Connector according to the API Manager version that you are using. See the 

Reviewer: I think if we move this section into "Deploy WSO2 Open Banking with Docker" and request them to download only if the docker tags are greater than the mentioned versions

Reviewer 2: @DivyaPremanantha How about if we do that change in here ?? because downloading IS connector is common for both  **Deploy WSO2 Open Banking with Docker** and **Deploy WSO2 Open Banking with Docker Compose**

Reviewer: The we have to move the note here to indicate only to download if the U2 update level is higher than the mentioned level

Author: Fixed in 398400c3b4064e997829dc96906d1d09acc15fbe.

1. Go to the `docker-compose` directory inside the `<OB_DOCKER_HOME>` directory.

    ```shell
    cd <OB_DOCKER_HOME>/docker-compose
    ```

2. Select the `docker-compose` setup that you wish to use and navigate into it.

    ```shell 
    cd obam-with-obiam
    ```       
   
    !!! note
        Hereafter, the above-mentioned directory will be referred to as `<DOCKER_COMPOSE_HOME>`.    

3. By default, `docker-compose` is configured for the Docker images of the WSO2 Open Banking Accelerator which are based 
   on WSO2 API Manager 4.2.0 and WSO2 Identity Server 6.0.0. 
   Follow the additional instructions below only if you are using Docker images of WSO2 Open Banking solution with any other base product versions:

    1. Open the `docker-compose.yml` file in the `<DOCKER_COMPOSE_HOME>` directory.
    2. Change the base product versions of `image` names.
    3. Go to `services` > `mysql` > `volumes`.
    4. Change the SQL script according to your base product versions. You can find the respective SQL script 
    according to your base product version [here](https://github.com/wso2/docker-open-banking/tree/master/docker-compose/mysql/scripts).

> 4. Follow the steps below only if you wish to run the Docker Compose setup using WSO2 UK Toolkit Docker Images or locally-built Docker images:

Reviewer: Shall we remove "WSO2 UK Toolkit" and mention "WSO2 Toolkit" instead

Author: Fixed in 398400c3b4064e997829dc96906d1d09acc15fbe.

    1. Build Docker images using Docker resources available [here](https://github.com/wso2/docker-open-banking/tree/master/dockerfiles).
    2. Remove the `docker.wso2.com/` prefix from the `image` name in the `docker-compose.yml` and change the image name to the image name of the locally-built image.

5. Volume mount the IS connector on the `obiam` container.

    !!!info
        This step is required based on the WSO2 Identity Server version and the update level you are using:

>         - For WSO2 Identity Server 5.11 as the base product, follow the instructions below if you are using Docker images with the tag **docker.wso2.com/wso2-obiam:3.0.0.76-is5.11.0.239 or above**.

Reviewer: A product version should have 3 digits, IS 5.11.0. Check other places also

Author: Fixed in 398400c3b4064e997829dc96906d1d09acc15fbe.

        - For WSO2 Identity Server 6.0 as the base product, follow the instructions below if you are using Docker images with the tag **docker.wso2.com/wso2-obiam:3.0.0.76-is6.0.0.71 or above**.

    1. Go to the `volumes` section of the `obiam` service in the `docker-compose.yml` in the `<DOCKER_COMPOSE_HOME>` directory.
    2. Change the root directory path of the extracted WSO2 IS Connector with `<IS_CONNECTOR_HOME>`.

7. Deploy the solution by executing the following command:

    ```shell
    docker-compose up
    ```

## Deploy WSO2 Open Banking with Docker

This section explains how to set up the solution using WSO2 Open Banking Docker images.

### Set up Network

- Create a network.
    
    ```shell
    docker network create -d bridge ob-network
    ```

### Set up Database Container

1. Pull the MySQL image.

    ```shell
    docker pull mysql:8.0.32
    ```
   
2. Run the MySQL Docker container.

    ```shell 
    docker run --network ob-network --name mysql -e MYSQL_ROOT_PASSWORD=root  -d mysql:8.0.32
    ```

> 3. Choose the SQL script according to your base product versions and copy that SQL script to the MySQL container.

Reviewer: We'll mention for APIM 4.2.0 and IS 6.0.0 the below script can be executed

Author: Fixed in 398400c3b4064e997829dc96906d1d09acc15fbe.

    ``` 

4. If you wish to use WSO2 Open Banking Business Intelligence Accelerator, copy the `setup-reporting-databases.sql` to the MySQL container.

    ```shell 
    docker cp <OB_DOCKER_HOME>/docker-compose/mysql/scripts/setup-reporting-databases.sql  mysql:/setup-reporting-databases.sql
    ``` 

5. Log in to the MySQL container.

    ```shell 
    docker exec -it mysql mysql -uroot -proot
    ``` 

6. Source the copied SQL script.

    ```  
    mysql> source setup.sql;
   
    mysql> source setup-reporting-databases.sql;
    ``` 

7. Update the MySQL connection limit.

    ```  
    mysql> set global max_connections = 1000;
    ``` 

8. Update the `USER` and `openbank_apimgtdb.SP_METADATA` tables:

    ``` 
    mysql> ALTER USER 'root'@'%' IDENTIFIED WITH mysql_native_password BY 'root';
   
    mysql> ALTER TABLE openbank_apimgtdb.SP_METADATA MODIFY VALUE VARCHAR(7500);
    ```

### Set up Open Banking Identity Server with Docker

1. Pull the Open Banking Identity Server image from [WSO2 Docker Repositories](https://docker.wso2.com/tags.php?repo=wso2-obiam).

    ```shell
    docker pull docker.wso2.com/wso2-obiam:3.0.0.0-is6.0.0.0
    ```

> 2. Deploy the Identity Server image.

Reviewer: Rephase this to "volume mount the IS connector and run the Identity Server image."

Author: Fixed in 398400c3b4064e997829dc96906d1d09acc15fbe.

    ```

    !!! note
        Here, `<IS_CONNECTOR_HOME>` refers to the root directory path of the extracted WSO2 IS Connector.
   
    !!!info
        The IS Connector is required based on the WSO2 Identity Server version and the update level you are using:

        - For WSO2 Identity Server 5.11 as the base product, follow the instructions below if you are using Docker images with the tag **docker.wso2.com/wso2-obiam:3.0.0.76-is5.11.0.239 or above**.
        - For WSO2 Identity Server 6.0 as the base product, follow the instructions below if you are using Docker images with the tag **docker.wso2.com/wso2-obiam:3.0.0.76-is6.0.0.71 or above**.

### Set up Open Banking API Manager with Docker

1. Pull the Open Banking API Manager image from [WSO2 Docker Repositories](https://docker.wso2.com/tags.php?repo=wso2-obam).

    ```shell 
    docker pull docker.wso2.com/wso2-obam:3.0.0.0-am4.2.0.0
    ```

> 2. Deploy the Open Banking API Manager image.

Reviewer: Rephase deploy to run

Author: Fixed in 398400c3b4064e997829dc96906d1d09acc15fbe.

1. Pull the Open Banking API Manager image from [WSO2 Docker Repositories](https://docker.wso2.com/tags.php?repo=wso2-obam).

     ```
     https://obiam:9446/api/openbanking/dynamic-client-registration
     ```
    ```shell 
    docker pull docker.wso2.com/wso2-obam:3.0.0.0-am4.2.0.0
    ```

13. When configuring the Key Manager, set the value of `IS_HOST` as `obiam`. For example,
2. Deploy the Open Banking API Manager image.

     ``` 
     https://obiam:9446/keymanager-operations/dcr/register
     ```
    ```shell
    docker run -p 9443:9443 -p 8243:8243 -p 8280:8280 --network ob-network --name obam docker.wso2.com/wso2-obam:3.0.0.0-am4.2.0.0
    ```

### Set up Open Banking API Manager with Docker
### Set up Open Banking Business Intelligence with Docker

1. Pull the Open Banking API Manager Image from [WSO2 Docker Repositories](https://docker.wso2.com/tags.php?repo=wso2-obam).
1. Pull the Open Banking Business Intelligence image from [WSO2 Docker Repositories](https://docker.wso2.com/tags.php?repo=wso2-obiam).

    ``` 
    docker pull docker.wso2.com/wso2-obam
    ```shell
    docker pull docker.wso2.com/wso2-obbi:3.0.0.0-si4.2.0.0
    ```

> 2. Deploy the API Manager Image.

Reviewer: Rephase deploy to run

Reviewer 3: @Chathuratharushka shall we change this deploy keyword in docker images (not docker compose) to `run` in readme files too?, I'll change in OBBI readme file

Reviewer 2: @hasithakn ack.

> 2. Deploy the Open Banking Business Intelligence image.

Reviewer: Rephase deploy to run

Author: Fixed in 398400c3b4064e997829dc96906d1d09acc15fbe.

    1. Open the `docker-compose.yml` file in the `<DOCKER_COMPOSE_HOME>` directory.
    2. Change the base product versions of `image` names.
    3. Go to `services` > `mysql` > `volumes`.
    4. Change the SQL script according to your base product versions. You can find the respective SQL script 
    according to your base product version [here](https://github.com/wso2/docker-open-banking/tree/master/docker-compose/mysql/scripts).

4. Follow the steps below only if you wish to run the Docker Compose setup using WSO2 UK Toolkit Docker Images or locally-built Docker images:

    1. Build Docker images using Docker resources available [here](https://github.com/wso2/docker-open-banking/tree/master/dockerfiles).
    2. Remove the `docker.wso2.com/` prefix from the `image` name in the `docker-compose.yml` and change the image name to the image name of the locally-built image.

5. Volume mount the IS connector on the `obiam` container.

    !!!info
        This step is required based on the WSO2 Identity Server version and the update level you are using:

>         - For WSO2 Identity Server 5.11, follow the instructions below if you are using **WSO2 Open Banking Identity Server Accelerator at U2 level 3.0.0.76 or above** and **WSO2 Identity Server at U2 level 5.11.0.239 or above**.

Author: Fixed in b16d9832f6eed555176959f9b2b0e8d501aaa6fd.

>         - For WSO2 Identity Server 6.0, follow the instructions below if you are using **WSO2 Open Banking Identity Server Accelerator at U2 level 3.0.0.76 or above** and **WSO2 Identity Server at U2 level 6.0.0.71 or above**.

Reviewer 3: ```suggestion
       For WSO2 Identity Server 6.0 as the base product, Follow the instructions below if you are using docker images **docker.wso2.com/wso2-obiam:3.0.0.76-is6.0.0.71 or above.**.
```

Author: Fixed in b16d9832f6eed555176959f9b2b0e8d501aaa6fd.

    ```  
    mysql> set global max_connections = 1000;
    ``` 

8. Update the `USER` and `openbank_apimgtdb.SP_METADATA` tables:

    ``` 
    mysql> ALTER USER 'root'@'%' IDENTIFIED WITH mysql_native_password BY 'root';
   
    mysql> ALTER TABLE openbank_apimgtdb.SP_METADATA MODIFY VALUE VARCHAR(7500);
    ```

### Set up Open Banking Identity Server with Docker

1. Pull the Open Banking Identity Server image from [WSO2 Docker Repositories](https://docker.wso2.com/tags.php?repo=wso2-obiam).

    ```shell
    docker pull docker.wso2.com/wso2-obiam:3.0.0.0-is6.0.0.0
    ```

2. Deploy the Identity Server image.

    ```shell
    docker run -it -p 9446:9446 --network ob-network --name obiam -v <IS_CONNECTOR_HOME>/dropins:/home/wso2carbon/wso2-artifact-volume/repository/components/dropins/ -v <IS_CONNECTOR_HOME>/webapps:/home/wso2carbon/wso2-artifact-volume/repository/deployment/server/webapps/ wso2-obiam:3.0.0.0-is6.0.0.0
    ```

    !!! note
        Here, `<IS_CONNECTOR_HOME>` refers to the root directory path of the extracted WSO2 IS Connector.
   
    !!!info
        The IS Connector is required based on the WSO2 Identity Server version and the update level you are using:

>         - For WSO2 Identity Server 5.11, follow the instructions below if you are using **WSO2 Open Banking Identity Server Accelerator at U2 level 3.0.0.76 or above** and **WSO2 Identity Server at U2 level 5.11.0.239 or above**.

Reviewer 3: lets refer the above suggested format.

Author: Fixed in b16d9832f6eed555176959f9b2b0e8d501aaa6fd.

   on WSO2 API Manager 4.2.0 and WSO2 Identity Server 6.0.0. 
   Follow the additional instructions below only if you are using Docker images of WSO2 Open Banking solution with any other base product versions:

    1. Open the `docker-compose.yml` file in the `<DOCKER_COMPOSE_HOME>` directory.
    2. Change the base product versions of `image` names.
    3. Go to `services` > `mysql` > `volumes`.
    4. Change the SQL script according to your base product versions. You can find the respective SQL script 
    according to your base product version [here](https://github.com/wso2/docker-open-banking/tree/master/docker-compose/mysql/scripts).

5. Follow the steps below only if you wish to run the Docker Compose setup using WSO2 UK Toolkit Docker Images or locally-built Docker images:

    1. Build Docker images using Docker resources available [here](https://github.com/wso2/docker-open-banking/tree/master/dockerfiles).
    2. Remove the `docker.wso2.com/` prefix from the `image` name in the `docker-compose.yml` and change the image name to the image name of the locally-built image.

> 6. Volume mount the IS connector on the `obiam` container.

Reviewer 3: Let's mention the u2 update level here, as only after below u2 levels we need to mount the IS connector,

If using IS 5.11 -> docker.wso2.com/wso2-obiam:3.0.0.76-is5.11.0.239
If using IS 6.0 -> docker.wso2.com/wso2-obiam:3.0.0.76-is6.0.0.71

Author: Fixed in f3cdd7f2d224b754e2ac59a513ae5b5cc5441edd.

    ```shell 
    docker cp <OB_DOCKER_HOME>/docker-compose/mysql/scripts/setup-reporting-databases.sql  mysql:/setup-reporting-databases.sql
    ``` 

5. Log in to the MySQL container.

    ```shell 
    docker exec -it mysql mysql -uroot -proot
    ``` 

6. Source the copied SQL script.

    ```  
    mysql> source setup.sql;
   
    mysql> source setup-reporting-databases.sql;
    ``` 

7. Update the MySQL connection limit.

    ```  
    mysql> set global max_connections = 1000;
    ``` 

8. Update the `USER` and `openbank_apimgtdb.SP_METADATA` tables:

    ``` 
    mysql> ALTER USER 'root'@'%' IDENTIFIED WITH mysql_native_password BY 'root';
   
    mysql> ALTER TABLE openbank_apimgtdb.SP_METADATA MODIFY VALUE VARCHAR(7500);
    ```

### Set up Open Banking Identity Server with Docker

1. Pull the Open Banking Identity Server image from [WSO2 Docker Repositories](https://docker.wso2.com/tags.php?repo=wso2-obiam).

    ```shell
    docker pull docker.wso2.com/wso2-obiam:3.0.0.0-is6.0.0.0
    ```

2. Deploy the Identity Server image.

    ```shell

>     docker run -it -p 9446:9446 --network ob-network --name obiam -v <IS_CONNECTOR_HOME>/dropins:/home/wso2carbon/wso2-artifact-volume/repository/components/dropins/ -v <IS_CONNECTOR_HOME>/webapps:/home/wso2carbon/wso2-artifact-volume/repository/deployment/server/webapps/ wso2-obiam:3.0.0.0-is6.0.0.0

Reviewer 3: Also let's mention IS connector is needed only after above mentioned updated levels

Author: Fixed in f3cdd7f2d224b754e2ac59a513ae5b5cc5441edd.

    ```

    !!! note
        Here, `<IS_CONNECTOR_HOME>` refers to the root directory path of the extracted WSO2 IS Connector.

### Set up Open Banking API Manager with Docker

1. Pull the Open Banking API Manager image from [WSO2 Docker Repositories](https://docker.wso2.com/tags.php?repo=wso2-obam).

    ```shell 
    docker pull docker.wso2.com/wso2-obam:3.0.0.0-am4.2.0.0
    ```

2. Deploy the Open Banking API Manager image.

    ```shell
    docker run -p 9443:9443 -p 8243:8243 -p 8280:8280 --network ob-network --name obam docker.wso2.com/wso2-obam:3.0.0.0-am4.2.0.0
    ```

### Set up Open Banking Business Intelligence with Docker

1. Pull the Open Banking Business Intelligence image from [WSO2 Docker Repositories](https://docker.wso2.com/tags.php?repo=wso2-obiam).

    ```shell
    docker pull docker.wso2.com/wso2-obbi:3.0.0.0-si4.2.0.0
    ```

2. Deploy the Open Banking Business Intelligence image.

    ```shell
    docker run -it -p 9712:9712 -p 9612:9612 -p 7712:7712 -p 7612:7612 -p 9092:9092 -p 9444:9444 --network ob-network --name obbi wso2-obbi:3.0.0.0-si4.2.0.0
    ```

> ### Configure the WSO2 Open Banking solution

Reviewer 3: ```suggestion
## Configure the WSO2 Open Banking solution
```

Author: Fixed in f3cdd7f2d224b754e2ac59a513ae5b5cc5441edd.

        ```toml
        [open_banking.data_publishing]
        enable = true
        ```
       
5. Place the modified `deployment.toml` files in the containers:

    ```shell
    docker cp <DESIRED_LOCATION_OF_IS_DEPLOYMENT>/deployment.toml obiam:/home/wso2carbon/wso2is-6.0.0/repository/conf/deployment.toml
    
    docker cp <DESIRED_LOCATION_OF_APIM_DEPLOYMENT>/deployment.toml obam:/home/wso2carbon/wso2am-4.2.0/repository/conf/deployment.toml
    ```
   
6. Restart the containers to apply the changes:

    ```shell
    docker restart obiam obam
    ```
   
7. Add the below entry mappings into the `/etc/hosts` file of your host machine.

    - `127.0.0.1 mysql`
    - `127.0.0.1 obam`
    - `127.0.0.1 obiam`

8. You can access the WSO2 Open Banking API Manager using a web browser via the following URLs:

    - `https://obam:9443/publisher`
    - `https://obam:9443/devportal`
    - `https://obam:9443/admin`
    - `https://obam:9443/carbon`

9. The API Gateway will be available on the following ports:

    - `https://obam:8243`
    - `http://obam:8280`

10. You can access the WSO2 Open Banking Identity Server using a web browser via the following URL:

>     - `https://obiam:9443/carbon`

Reviewer 2: shall we change the port number to 9446 ?  it should be 9446. I have added it as 9443 in the google doc by mistakenly.

Author: Fixed via 5a46f883070c2024459104734039bc097857d38d.

> ### Deploy APIs and Configure IS as Key manager

Reviewer 2: ```suggestion
## Deploy APIs and Configure IS as Key manager
```

Author: Fixed in f3cdd7f2d224b754e2ac59a513ae5b5cc5441edd.

1. If you are using HTTP/REST Endpoints when publishing the APIs, update the hostname of the endpoint as `obiam`. For example,

    ```
    https://obiam:9446/api/openbanking/dynamic-client-registration
    ```

2. If you are using insequence files when publishing the APIs, replace the hostname in the insequence file as follows:

    - `obam:9443`
    - `obiam:9446`

3. When configuring the Key Manager, set the value of `IS_HOST` as `obiam`. For an example,

    ```
    https://obiam:9446/keymanager-operations/dcr/register
    ```

> ## Download WSO2 Updates

Reviewer 3: I guess we can remove this section, When updating the docker images, we can get the latest image rather than running u2 script

Author: Removed in f3cdd7f2d224b754e2ac59a513ae5b5cc5441edd.

    ```shell
    docker pull docker.wso2.com/wso2-obbi:3.0.0.0-si4.2.0.0
    ```

2. Run the Open Banking Business Intelligence image.

    ```shell
    docker run -it -p 9712:9712 -p 9612:9612 -p 7712:7712 -p 7612:7612 -p 9092:9092 -p 9444:9444 --network ob-network --name obbi wso2-obbi:3.0.0.0-si4.2.0.0
    ```

### Set up Open Banking Identity Server with Docker

1. Pull the Open Banking Identity Server image from [WSO2 Docker Repositories](https://docker.wso2.com/tags.php?repo=wso2-obiam).

    ```shell
    docker pull docker.wso2.com/wso2-obiam:3.0.0.0-is6.0.0.0
    ```

2. Volume mount the IS connector and run the Identity Server image.

    ```shell
    docker run -it -p 9446:9446 --network ob-network --name obiam -v <IS_CONNECTOR_HOME>/dropins:/home/wso2carbon/wso2-artifact-volume/repository/components/dropins/ -v <IS_CONNECTOR_HOME>/webapps:/home/wso2carbon/wso2-artifact-volume/repository/deployment/server/webapps/ wso2-obiam:3.0.0.0-is6.0.0.0
    ```

    !!! note
        Here, `<IS_CONNECTOR_HOME>` refers to the root directory path of the extracted WSO2 IS Connector.
   

>     !!!info

Reviewer 2: let's remove this from here aswell.

Author: Fixed in 1041fbadad35af42da289f97ad8584491e28551c.

    1. Open the `docker-compose.yml` file in the `<DOCKER_COMPOSE_HOME>` directory.
    2. Change the base product versions of `image` names.
    3. Go to `services` > `mysql` > `volumes`.
    4. Change the SQL script according to your base product versions. You can find the respective SQL script 
    according to your base product version [here](https://github.com/wso2/docker-open-banking/tree/master/docker-compose/mysql/scripts).

4. Follow the steps below only if you wish to run the Docker Compose setup using WSO2 Toolkit Docker Images or locally-built Docker images:

    1. Build Docker images using Docker resources available [here](https://github.com/wso2/docker-open-banking/tree/master/dockerfiles).
    2. Remove the `docker.wso2.com/` prefix from the `image` name in the `docker-compose.yml` and change the image name to the image name of the locally-built image.

5. Volume mount the IS connector on the `obiam` container.

    !!!info
        This step is required based on the WSO2 Identity Server version and the update level you are using:

>         - For WSO2 Identity Server 5.11.0 as the base product, follow the instructions below if you are using Docker images with the tag **docker.wso2.com/wso2-obiam:3.0.0.76-is5.11.0.239 or above**.

Author: Fixed in 66575352429b5b4ba6b16474f9b7d88ffbdd74cf.
